### ISSUE TITLE
WASM as an installed app

### ISSUE THREAD

**Issue by JimHammond** [csharpfritz/csharp_with_csharpfritz#106]
Hi, Fritz! Can you talk about how when a user visits our page that has our WASM, we can immediately detect if that user has installed our WASM locally, and if not, how we can pop up a window in which they can click a button to install it? Any gotchas we should know about? Thanks!

**Comment by csharpfritz** [comment]
Why pop up a window, and instead just update the download?  Blazor uses
this model to update your WASM content and store it in your browser's
Application Cache


Jeff

**Comment by JimHammond** [comment]
I would like the user to be able to install my WASM locally as an icon on the screen as if it were an app, and never have to visit my web page again. I think I was mistaken though about wanting to pop up a window when the user visits my page. I think I would actually want to do that when the user starts my WASM for the first time. I am still designing the app, so I haven't actually built any of it yet.

**Comment by csharpfritz** [comment]
Oh terrific!  Take a look at using Blazor as a PWA. There’s a great mechanism for that 



Jeff

**Comment by csharpfritz** [comment]
I'll cover the configuration and installation of a Blazor application as a PWA.

**Comment by JimHammond** [comment]
Thanks! A few weeks ago, before I got sick, I created a PWA Blazor WASM, and in Chrome browsers it would correctly display an icon in the address bar that would let the user install the WASM locally so it would appear to the user as if it were just another app, which was great except for two things: 1) It only appeared in Chrome browsers and not in any other browsers, and 2) It was not at all obvious. It would have been overlooked by 99% of users.
